script.js

```diff
@@ -0,0 +1,121 @@
+let loop = function(times = 0, callback = null) {
+    if (typeof callback != 'function') throw new TypeError('Callback required');
+    for (let i = 0; i < times; i++) {
+        callback();
+    }
+};
+let calculateArea = function(figure, ...parameters) {
+    parameters.forEach( arg => {
+        if (!Number.isInteger(arg)) throw new TypeError('Arguments must be integer');
+    });
+    let [a, b, c] = [parameters[0], parameters[1], parameters[2]]; // не хотел открывать в эту область видимости(для треугольника — 3 стороны, для квадрата — 1), но кучу строк введения переменных не оно, 'оптимизация' роли не сыграет.
+    let area;
+    switch (figure) {
+        case 'triangle':
+        let hpr = (a + b + c) / 2;
+        area = Math.sqrt(hpr * (hpr-a) * (hpr-b) * (hpr-c));
+        return { area: area, figure: figure, input: { sideA: a, sideB: b, sideC: c }};
+        case 'square':
+        area = a ** 2;
+        return { area: area, figure: figure, input: {sideA: a}};
+        case 'rectangle':
+        area = a * b;
+        return { area: area, figure: figure, input: {sideA: a, sideB: b}};
+        default:
+        throw new Error("Wrong figure");
+    }
+};
+class Human {
+    constructor(name, age, dateOfBirth) {
+        this.name = name;
+        this.age = age;
+        this.dateOfBirth = dateOfBirth;
+    }
+    displayInfo() {
+        return `Name: ${this.name}, Age: ${this.age}, Date of birth: ${this.dateOfBirth}`;
+    }
+}
+class Employee extends Human {
+    constructor(name, age, dateOfBirth, salary, department) {
+        super(name, age, dateOfBirth);
+        this.salary = salary;
+        this.department = department;
+    }
+    displayInfo() {
+        return `${super.displayInfo()}, Salary: ${this.salary}, Department: ${this.department}`;
+    }
+}
+class Manager extends Employee {
+    constructor(name, age, dateOfBirth, salary, department, allTheDevs) {
+        super(name, age, dateOfBirth, salary, department);
+        if (!Array.isArray(allTheDevs) || allTheDevs == undefined) throw new Error("allTheDevs argument must present and must be an array");
+        this.allTheDevs = allTheDevs;
+    }
+    getDevs() {
+        let string = "";
+        this.allTheDevs.forEach((arg, i) => {
```

Comment:
```suggestion
        this.allTheDevs.forEach((developer, i) => {
```
можно еще dev, так просто легче читать;)